luarules/gadgets/unit_area_timed_damage.lua

```diff
@@ -12,484 +12,229 @@ function gadget:GetInfo()
 
-local TimedDamageWeaponsNames = {
-    ['raptor_land_assault_acid_t2_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 100,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_allterrain_arty_acid_t2_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 200,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_allterrain_arty_acid_t4_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 200,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_air_bomber_acid_t2_v1_acidbomb'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 100,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_land_swarmer_acids_t2_v1_acidspit'] = {
-        ceg = "acid-area-75",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 40,
-        range = 75,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_allterrain_swarmer_acid_t2_v1_acidspit'] = {
-        ceg = "acid-area-75",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 40,
-        range = 75,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_allterrain_assault_acid_t2_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 100,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_turret_acid_t2_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 200,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_turret_acid_t3_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 200,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_turret_acid_t4_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 200,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_matriarch_acid_acidgoo'] = {
-        ceg = "acid-area-75",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 40,
-        range = 75,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_matriarch_acid_spike_acid_blob'] = {
-        ceg = "acid-area-75",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 40,
-        range = 75,
-        resistance = "_RAPTORACID_",
-    },
+local function AddTimedAreaDef(id, def)
+	if not def.customParams or not def.customParams.area_timed_damage == true then return end
```

Comment:
```suggestion
	if not def.customParams.area_timed_damage then return end
```
All unit defs have a customparams table injected during preprocessing, it does not need to be nil checked.

Reply:
This function handles weapon defs, also, which I didn't think had customparams injeted. Still, this guard belongs outside the function.

Reply:
Injecting during preprocessing has nothing to do with it because it's a different table.